go-controller/pkg/metrics/metrics.go

```diff
@@ -122,9 +122,9 @@ func getCoverageShowOutputMap(component string) (map[string]string, error) {
 
 // coverageShowMetricsUpdater updates the metric
 // by obtaining values from getCoverageShowOutputMap for specified component.
-func coverageShowMetricsUpdater(component string) {
+func coverageShowMetricsUpdater(component string, metricsScrapeInterval int) {
 	for {
-		time.Sleep(30 * time.Second)
+		time.Sleep(time.Duration(metricsScrapeInterval) * time.Second)
```

Comment:
...and then you don't need to pass the value into these functions, because they can just pull it from `config.Whatever.MetricsScrapInterval`

Reply:
@danwinship this will not work for the `ovs-exporter` subcommand of ovn-kube-util. The OVS metrics are exported by the ovn-kube-util process running in a container and it is invoked using the `ovs-exporter` subcommand, like so:

`/usr/bin/ovn-kube-util --loglevel=4 ovs-exporter --metrics-interval 30 --metrics-bind-address 10.8.51.51:9310`

The ovn-kube-util doesn't call InitConfig(), and it has its own set of options (different from `ovnkube`). Hence, passing of that option everywhere. 

Perhaps, for OVS metrics alone we should pass the value and for OVN-K8s and OVN metrics we should use the `config.whatever` like you say.

Reply:
yeah, the `ovn-kube-util` version would have to do it differently, but for `ovn-kubernetes` I'd really like to have all of the configuration-y command-line options available as config file options too